lib/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
-add_library(disccord SHARED client.cpp)
+set(SOURCE_FILES client.cpp)
+
+add_library(disccord SHARED ${SOURCE_FILES})
```

Comment:
Ah yes, you can do that with something like that:
```cmake
set(DISCCORD_BUILD_TESTS ON) # this sets tests as on

if(DISCCORD_BUILD_TESTS)
    add_subdirectory(test)
endif()
```

and in the "test" subdirectory you create a cmakelists like

```cmake
set(TEST_FILES testfile1.cc testfile2.cc) # as many files as you want

add_executable(disccord_test ${TEST_FILES})
```

then when you build with tests, cmake'll generate a test thing and you can run it. It's a simplified example but it ain't too hard.